modules/API/API.cpp

```diff
@@ -474,10 +475,11 @@ void API::init() {
         });
 
         std::string cmd_set_limit = cmd_base + "set_limit_amps";
-        this->mqtt.subscribe(cmd_set_limit, [&evse](const std::string& data) {
+        this->mqtt.subscribe(cmd_set_limit, [&evse, &r_evse_manager_energy_sink = this->r_evse_manager_energy_sink,
+                                             evse_id](const std::string& data) {
             try {
                 const auto external_limits = get_external_limits(data, false);
-                evse->call_set_external_limits(external_limits);
+                r_evse_manager_energy_sink.at(evse_id - 1)->call_set_external_limits(external_limits);
```

Comment:
Won't for `evse_id = 1`, `evse_id - 1` access the `grid_connection` evse manager energy sink here?